bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.net;
+
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Some utility functions related to network interfaces etc.
+ *
+ * @author Markus Rathgeb - Initial contribution and API
+ */
+public class NetUtil {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetUtil.class);
+
+    private NetUtil() {
+    }
+
+    /**
+     * Get the first candidate for a local IPv4 host address (non loopback, non localhost).
+     */
+    public static String getLocalIpv4HostAddress() {
```

Comment:
somehow this looks weird to me, IMHO getLocalIPv4HostAddress is more brain-friendly, but that could just be personal preference...